wrapper/src/main/java/software/amazon/jdbc/plugin/limitless/LimitlessConnectionPlugin.java

```diff
@@ -177,10 +172,7 @@ private Connection connectInternalWithDialect(
           new Object[] {selectedHostSpec != null ? selectedHostSpec.getHost() : "null"}));
     } catch (SQLException e) {
       LOGGER.warning(Messages.get("LimitlessConnectionPlugin.errorSelectingRouter", new Object[] {e.getMessage()}));
-      if (conn == null  || conn.isClosed()) {
-        conn = connectFunc.call();
-      }
-      return retryConnectWithLeastLoadedRouters(limitlessRouters, props, conn, hostSpec);
+      return retryConnectWithLeastLoadedRouters(limitlessRouters, props, connectFunc.call(), connectFunc, hostSpec);
```

Comment:
wondering in these cases where there is no connection object already instantiated, if i should just leave the connection param to be null.